cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1158,6 +1160,18 @@ def configure_apiserver(etcd_connection_string, leader_etcd_version):
         admission_control.remove('Initializers')
     api_opts['admission-control'] = ','.join(admission_control)
 
+    if get_version('kube-apiserver') > (1, 6) and \
+       hookenv.config('enable-metrics'):
+        api_opts['requestheader-client-ca-file'] = ca_cert_path
+        api_opts['requestheader-allowed-names'] = 'client'
+        api_opts['requestheader-extra-headers-prefix'] = 'X-Remote-Extra-'
+        api_opts['requestheader-group-headers'] = 'X-Remote-Group'
+        api_opts['requestheader-username-headers'] = 'X-Remote-User'
+        api_opts['proxy-client-cert-file'] = client_cert_path
+        api_opts['proxy-client-key-file'] = client_key_path
+        api_opts['enable-aggregator-routing'] = 'true'
+        api_opts['client-ca-file'] = ca_cert_path
```

Comment:
Can we just always set `client-ca-file` (and maybe the rest of these) regardless of the configuration of `enable-metrics`? Would it hurt anything?

I'm asking because, if we always set `client-ca-file`, it would provide a convenient fix for https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/515.

Reply:
@Cynerva I have no problem with that, I was just trying to reduce the scope of the change to only have an impact when metrics were enabled. Is this a request to make that change or just start a conversation?

Reply:
Just starting a conversation, sorry for not being clear on that. I don't think anything needs to change in this PR, it looks solid and I don't want to expand the scope of it without good reason. We can address https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/515 in a standalone PR later :)